src/nspawn/nspawn.c

```diff
@@ -2162,6 +2163,57 @@ static int setup_boot_id(void) {
         return mount_nofollow_verbose(LOG_ERR, NULL, to, NULL, MS_BIND|MS_REMOUNT|MS_RDONLY|MS_NOSUID|MS_NOEXEC|MS_NODEV, NULL);
 }
 
+static int get_fuse_version(uint32_t *major, uint32_t *minor) {
+        int r;
+        ssize_t n;
+        _cleanup_close_ int fd;
+        _cleanup_(umount_and_rmdir_and_freep) char *dir;
+        _cleanup_free_ char *opts = NULL;
+        union {
+                char unstructured[FUSE_MIN_READ_BUFFER];
+                struct {
+                        struct fuse_in_header header;
+                        /* Don't use <linux/fuse.h>:`struct fuse_init_in` because a newer fuse.h might give
+                         * us a bigger struct than what an older kernel actually gives us, and that would
+                         * break our short-read checks. */
+                        struct {
+                                uint32_t major;
+                                uint32_t minor;
+                        } body;
+                } structured;
+        } request;
+
+        /* Get a FUSE handle. */
+        fd = open("/dev/fuse", O_RDWR);
+        if (fd < 0)
+                return log_warning_errno(errno, "Disabling FUSE: Failed to open /dev/fuse: %m");
+        r = mkdtemp_malloc("/tmp/nspawn-fuse-XXXXXX", &dir);
+        if (r < 0)
+                return log_warning_errno(r, "Disabling FUSE: Failed to create temporary directory: %m");
+        if (asprintf(&opts, "fd=%i,rootmode=40000,user_id=0,group_id=0", fd) < 0)
+                return log_oom();
+        if (mount("nspawn-fuse", dir, "fuse.nspawn", 0, opts) < 0)
+                return log_warning_errno(r, "Disabling FUSE: Failed to mount test handle: %m");
+
+        /* Read a request from the FUSE handle. */
+        n = read(fd, &request.unstructured, sizeof request);
+        if (n < 0)
+                return log_warning_errno(errno, "Disabling FUSE: Failed to read /dev/fuse: %m");
+        if ((size_t) n < sizeof request.structured.header ||
+            (size_t) n < request.structured.header.len)
+                return log_warning_errno(-1, "Disabling FUSE: Failed to read /dev/fuse: Short read");
```

Comment:
please do not make up numeric errors such as -1. use SYNTHETIC_ERRNO(EIO) or something like that, to generate proper errno style errors

Reply:
Fixed.